### PR TITLE
Resolve Promise after HTTP fallback request returns

### PR DESCRIPTION
**Pull request by akoenig**
This PR contains a change in the HTTP fallback implementation. The `Promise` will resolve when there is a response from the HTTP response. Furthermore, it contains some minor `chory` things, like switching to new TypeScript 2.x compiler and removing `typings` from the project.

### CODE REVIEW

File: scripts/test [hunk 1]
```diff
@@ -4,7 +4,6 @@
 # Script for executing the test suite
 #
 
-npm run typings
 npm run compile
 
 mocha -- $(find dist -name '*.spec.js')
```
Comment (ansgarS) [inline scripts/test]: istanbul + source-map support?

File: src/lib/index.ts [hunk 2]
```diff
@@ -25,13 +23,14 @@ export interface IRequest {
...
-    const store = createStore({reducer, middlewares});
+    const store = createStore({ reducer, middlewares });
 
     const schedule = (manifest: IRequest) => {
         store.dispatch(
-            add(assign({}, manifest, {
+            add({
                 id: uuid(),
-            }))
+                ...manifest,
```
Comment (ansgarS) [inline src/lib/index.ts]: If manifest owns an ID, ```id``` would be overwritten - intended?
If not, you have to change the order ```{ ...manifest, id: uuid() }```

File: src/lib/sagas/add.ts [hunk 3]
```diff
@@ -7,19 +7,19 @@
...
 
 import {
     ADD,
 } from "../constants/actions";
 
-const dbg: debug.Debugger  = debug("@stylelounge/http-queue:sagas:add");
+const dbg: debug.IDebugger = debug("@stylelounge/http-queue:sagas:add");
 
-function * worker(action: Action<IManifest>): any {
+function* worker(action: Action<IManifest>): any {
```
Comment (ansgarS) [inline src/lib/sagas/add.ts]: Is this really a generator-function?

File: src/lib/sagas/remove.ts [hunk 4]
```diff
@@ -7,33 +7,30 @@
...
-import {
-    REMOVE,
-} from "../constants/actions";
+import { REMOVE } from "../constants/actions";
 
-const dbg: debug.Debugger  = debug("@stylelounge/http-queue:sagas:remove");
+const dbg: debug.IDebugger = debug("@stylelounge/http-queue:sagas:remove");
 
-function * worker(action: Action<number>): any {
+function* worker(action: Action<number>): any {
```
Comment (ansgarS) [inline src/lib/sagas/remove.ts]: same here

File: tsconfig.json [hunk 5]
```diff
@@ -1,29 +1,18 @@
...
-        "pretty": true,
-        "noImplicitUseStrict": true,
+        "declaration": true,
         "rootDir": "./src",
         "sourceRoot": "./src",
-        "outDir": "./dist"
+        "outDir": "./dist",
+        "forceConsistentCasingInFileNames": true,
+        "pretty": true,
+        "noImplicitAny": true
```
Comment (ansgarS) [inline tsconfig.json]: Don't you need ```TypeRoots```?

File: src/lib/index.ts (hunk 2 above)
Comment (akoenig) [inline src/lib/index.ts]: `IRequest` does not contain an `id`. Therefore, this is okay :)

File: src/lib/sagas/add.ts (hunk 3 above)
Comment (akoenig) [inline src/lib/sagas/add.ts]: Yap :)

File: tsconfig.json (hunk 5 above)
Comment (akoenig) [inline tsconfig.json]: Nope, we don't need own types here. TypeScript uses `@types` as default TypeRoot internally.

File: scripts/test (hunk 1 above)
Comment (akoenig) [inline scripts/test]: hmm ... hmm ... Coverage reports would be pretty bad :) – The main part in this case is to test the reducer and this part is covered.

**Pull request merged**